### PR TITLE
Improve httpclient with SNI support

### PR DESCRIPTION
**Pull request by xbiansf**
Improve http client from potential client leak and adding verification test for Server name indicator in SSL.

### CODE REVIEW

**Comment by rjmazzeo** [comment]
Nice job. This coupled with Jae's edit should really solidy http handling.

File: src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java [hunk 1]
```diff
@@ -73,80 +74,68 @@
...
-                        response = client.execute(head);
-                    } finally {
-                        response.close();
-                    }
+            CredentialsProvider credentialsprovider = new BasicCredentialsProvider();
+            credentialsprovider.setCredentials(scope, credentials);
+            httpClientBuilder.setDefaultCredentialsProvider(credentialsprovider).build();
+        }
+
+        try (CloseableHttpClient httpClient = httpClientBuilder.build()) {
```
Comment (jjangsfdc) [inline src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java]: Does it dispose/close itself after the Try block?

Comment (xbiansf) [inline src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java]: Yes.  All the try resource block will call close whenever there is any exception.  If there is any exception inside close(), that exception will be suppressed which is what we want most of the time. This also releases the resource from previous HttpClients.createDefault() as well by using the builder pattern.

Comment (jjangsfdc) [inline src/main/java/com/salesforce/dataloader/client/DefaultSimplePost.java]: Nice. Please make sure to run the integration tests before we pull in the change.

**Pull request merged**